specification/sbi_atee.adoc

```diff
@@ -10,8 +39,1296 @@ shown in the figure above, the TEE Runtime (operating in VS-mode) supports
+`sbi_tee_interrupt_init_tvm_aia()`. No two VCPUs may share the same `tvm_vcpu_imsic_gpa`.
+
+This can be called only after `sbi_tee_interrupt_init_tvm_aia()` and before `sbi_tee_host_finalize_tvm()`.
+All VCPUs in an AIA-enabled TVM must have their IMSIC configuration set prior to calling
+`sbi_tee_host_finalize_tvm()`.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_interrupt_set_tvm_aia_cpu_imsic_addr_errors]
+.TEE Interrupt Set TVM AIA CPU IMSIC Addr
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `tvm_vcpu_imsic_gpa` was invalid.
+| SBI_ERR_INVALID_PARAM   | `tvm_guest_id` or `tvm_vcpu_id` were invalid, or
+                            the TVM wasn't in the `TVM_INITIALIZING` state.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Interrupt Convert AIA IMSIC (FID #2)
+[source, C]
+-------
+struct sbiret sbi_tee_interrupt_convert_aia_imsic(unsigned long imsic_page_addr);
+-------
+
+Starts the process of converting the non-confidential guest interrupt file at
+`imsic_page_addr` for use with a TVM. This must be followed by calls to `sbi_tee_host_global_fence()`
+and `sbi_tee_host_local_fence()` before the interrupt file can be assigned to a TVM.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_aia_tvm_convert_imsic_errors]
+.TEE Interrupt Convert AIA IMSIC
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `imsic_page_addr` was invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Interrupt Reclaim TVM AIA IMSIC (FID #3)
+[source, C]
+-------
+struct sbiret sbi_tee_interrupt_reclaim_tvm_aia_imsic(unsigned long imsic_page_addr);
+-------
+
+Reclaims the confidential TVM interrupt file at `imsic_page_addr`. The interrupt file
+must not currently be assigned to a TVM.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_reclaim_tvm_aia_imsic_errors]
+.TEE Interrupt Reclaim TVM AIA IMSIC
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `imsic_page_addr` was invalid.
+| SBI_ERR_INVALID_PARAM   | The memory is still assigned to a TVM.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+== TEE Guest Extension (EID 0x54454547)
+The TEE Guest extension supplements the TEE Host extension, and TVMs to communicate with TSM. A typical
+usecase for this extension is to relay information to the host.
```

Comment:
Probably need a blurb here about how TEE-Guest calls cause an exit to the host with scause set to ECALL, a6 set to FID, a0-a5 set to ECALL args, etc.